crates/sui-framework/src/natives/object_runtime/object_store.rs

```diff
@@ -38,6 +40,10 @@ struct Inner<'a> {
     // cached objects from the resolver. An object might be in this map but not in the store
     // if it's existence was queried, but the value was not used.
     cached_objects: BTreeMap<ObjectID, Option<MoveObject>>,
+    // This sets a limit on the number of cached objects
+    max_num_cached_objects: u64,
```

Comment:
any reason to not just keep the protocol config? Just wondering the reasoning one way or another

Reply:
Only because ProtocolConfig is a large struct.